DHFPROD-6645: Each after match save, call validation API and display …

brucean52 opened the pull request:
…warning

### Description

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [ ] JIRA_ID included in all the commit messages
- [ ] PR title is in the format JIRA_ID:Title
- [ ] Rebase the branch with upstream
- [ ] Squashed all commits into a single commit
- [ ] Code passes ESLint tests
- [ ] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests

ngodugu-marklogic commented:
@brucean52 I see scripts mappingWithCustomHeader.spec.tsx and mappingWithInterceptorsAndCustomHook.spec.tsx failing consistently on this PR. Can you please run these tests in local and confirm they are going fine.

brucean52 commented:
> @brucean52 I see scripts mappingWithCustomHeader.spec.tsx and mappingWithInterceptorsAndCustomHook.spec.tsx failing consistently on this PR. Can you please run these tests in local and confirm they are going fine.


> @brucean52 I see scripts mappingWithCustomHeader.spec.tsx and mappingWithInterceptorsAndCustomHook.spec.tsx failing consistently on this PR. Can you please run these tests in local and confirm they are going fine.

I rebased the PR.

ngodugu-marklogic commented:
@brucean52 When we create a new map step, the step details page for the new steps gets displayed immediately after we click save.Please refer https://drive.google.com/file/d/1SArmWIRYKEYNSeI56y9QGEFuBZUcEJtX/view .I think this is not happening in the PR due to which the scripts are failing. Can you please verify it from your end

ngodugu-marklogic reviewed (changes requested):
@brucean52 When we create a new map step, the step details page for the new steps gets displayed immediately after we click save.Please refer https://drive.google.com/file/d/1SArmWIRYKEYNSeI56y9QGEFuBZUcEJtX/view .I think this is not happening in the PR due to which the scripts are failing. Can you please verify it from your end

Sanjeevani19 commented:
> @brucean52 When we create a new map step, the step details page for the new steps gets displayed immediately after clicking save.Please refer https://drive.google.com/file/d/1SArmWIRYKEYNSeI56y9QGEFuBZUcEJtX/view .I think this is not happening in the PR due to which the scripts are failing. Can you please verify it from your end

@brucean52 This happens in both match and merge tabs too, Please verify those scenarios too

brucean52 commented:
> > @brucean52 When we create a new map step, the step details page for the new steps gets displayed immediately after clicking save.Please refer https://drive.google.com/file/d/1SArmWIRYKEYNSeI56y9QGEFuBZUcEJtX/view .I think this is not happening in the PR due to which the scripts are failing. Can you please verify it from your end

This PR did not make any changes to the mapping step component, but I rebased and tested and I couldn't reproduce it.

> @brucean52 This happens in both match and merge tabs too, Please verify those scenarios too

I rebased the PR to develop and not reproduce this scenario. https://drive.google.com/file/d/1nT2aciv8d09W1i8Z7qv1-jMMk7TBEdzI/view

Sanjeevani19 reviewed (commented):
@brucean52  when adding a new match step, if I don't give a source query name and click on save the modal closes, since source query is a required field we should display a error message right ? https://drive.google.com/file/d/1DHmLxCDQI1g3lpZ3uQlXwOAK3CwVz2M6/view

ngodugu-marklogic commented:
> > > @brucean52 When we create a new map step, the step details page for the new steps gets displayed immediately after clicking save.Please refer https://drive.google.com/file/d/1SArmWIRYKEYNSeI56y9QGEFuBZUcEJtX/view .I think this is not happening in the PR due to which the scripts are failing. Can you please verify it from your end
> 
> This PR did not make any changes to the mapping step component, but I rebased and tested and I couldn't reproduce it.
> 
> > @brucean52 This happens in both match and merge tabs too, Please verify those scenarios too
> 
> I rebased the PR to develop and not reproduce this scenario. https://drive.google.com/file/d/1nT2aciv8d09W1i8Z7qv1-jMMk7TBEdzI/view

@brucean52 Was able to reproduce the issue. https://drive.google.com/file/d/1VGRccM2kBx7kP6e_57ALjrJp-IX4hyVn/view

Sanjeevani19 reviewed (commented):
@brucean52 When I tested this morning I was not able to see the warnings which were mentioned in the story, this was working before but now the warnings don't show up, Please look into that too
